### PR TITLE
Child workflow continue as new cancellation test

### PR DESCRIPTION
**Pull request by vkoby**
## Checklist

1. Closes issue: 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: temporal-sdk/src/test/java/io/temporal/workflow/childWorkflowTests/ChildWorkflowContinueAsNewCancellationTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    WorkflowStub client = testWorkflowRule.newUntypedWorkflowStub("TestWorkflow");
+    WorkflowExecution execution = client.start(ChildWorkflowCancellationType.TRY_CANCEL);
+    testWorkflowRule.sleep(Duration.ofSeconds(3));
+    client.cancel();
+    try {
+      client.getResult(String.class);
+      Assert.fail("unreachable");
+    } catch (WorkflowFailedException e) {
+      assertTrue(e.getCause() instanceof CanceledFailure);
+      System.out.println("CANCELLED!");
```
Comment (vitarb) [inline temporal-sdk/src/test/java/io/temporal/workflow/childWorkflowTests/ChildWorkflowContinueAsNewCancellationTest.java:63]: Remove?

File: temporal-sdk/src/test/java/io/temporal/workflow/childWorkflowTests/ChildWorkflowContinueAsNewCancellationTest.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+          .build();
+
+  @Test
+  public void testChildWorkflowContinueAsNewCancellationTest() throws InterruptedException {
+    count = new AtomicInteger(3);
+    WorkflowStub parentWorkflow =
+        testWorkflowRule.newUntypedWorkflowStub("TestWorkflowCancellationType");
+    WorkflowExecution parentExecution =
+        parentWorkflow.start(ChildWorkflowCancellationType.WAIT_CANCELLATION_COMPLETED);
+    Thread.sleep(500);
```
Comment (Sushisource) [inline temporal-sdk/src/test/java/io/temporal/workflow/childWorkflowTests/ChildWorkflowContinueAsNewCancellationTest.java:60]: This is a pretty long explicit sleep to have in a test. It'd be better to have the workflows set some condition in the class that can be waited on.

File: temporal-sdk/src/test/java/io/temporal/workflow/childWorkflowTests/ChildWorkflowContinueAsNewCancellationTest.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  }
+
+  public static class TestParentWorkflowImpl implements TestWorkflowCancellationType {
+    @Override
+    public void execute(ChildWorkflowCancellationType cancellationType) {
+      NoArgsWorkflow child =
+          Workflow.newChildWorkflowStub(
+              NoArgsWorkflow.class,
+              ChildWorkflowOptions.newBuilder().setCancellationType(cancellationType).build());
+      child.execute();
```
Comment (Sushisource) [inline temporal-sdk/src/test/java/io/temporal/workflow/childWorkflowTests/ChildWorkflowContinueAsNewCancellationTest.java:109]: Based on issue writeup I think this WF needs to issue a cancel (good to test signal too) to its child. Worth clarifying with Wenquan

File: temporal-sdk/src/test/java/io/temporal/workflow/childWorkflowTests/ChildWorkflowContinueAsNewCancellationScopeTest.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      Assert.fail("unreachable");
+    } catch (WorkflowFailedException e) {
+      assertTrue(e.getCause() instanceof CanceledFailure);
+    }
+    testWorkflowRule.assertHistoryEvent(
+        parentExecution, EventType.EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_CANCELED);
+  }
+
+  @Test
+  public void testImposterWorkflowContinueAsNewCancellationTest() throws InterruptedException {
```
Comment (vkoby) [inline temporal-sdk/src/test/java/io/temporal/workflow/childWorkflowTests/ChildWorkflowContinueAsNewCancellationScopeTest.java:81]: @mfateev @vitarb, could you please take a one last look at this test? I feel like I'm so close to reproducing Wenquan's [scenario](https://github.com/temporalio/sdk-java/issues/501#issuecomment-862620117), but getting a InternalWorkflowTaskException.

**Pull request closed without merging**